lib/hyku_addons/engine.rb

```diff
@@ -319,11 +319,11 @@ def update_admin_set(attrs)
         def find
           return find_by_id if attributes[:id]
           return search_by_identifier if attributes[system_identifier_field].present?
-          return search_by_title if klass == AdminSet && attributes[:title].present?
+          return search_by_title_or_identifier if klass == AdminSet && attributes[:title].present?
         end
 
-        def search_by_title
-          AdminSet.where(title: Array(attributes[:title]).first).first
+        def search_by_title_or_identifier
+          AdminSet.where(title: Array(attributes[:title]).first).first || AdminSet.where(id: Array(attributes[:title]).first).first
```

Comment:
I like using `find_by` if I am only interested on one result, as it avoids the `first` and limits the DB results to 1 too.
```suggestion
          AdminSet.find_by(title: Array(attributes[:title]).first) || AdminSet.where(id: Array(attributes[:title]).first).first
```